File: backend/app/schemas/ingestion.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class IngestionRunAccepted(BaseModel):
    run_id: str
    status: str


class IngestionRunStatus(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    run_id: str
    status: str
    result_type: str | None = None
    started_at: datetime
    completed_at: datetime | None = None
    cursor_used: str | None = None
    cursor_advanced: bool
    candidate_dataset_id: str | None = None
    dataset_version_id: str | None = None
    records_received: int | None = None
    failure_reason: str | None = None


class CurrentDataset(BaseModel):
    source_name: str
    dataset_version_id: str
    updated_at: datetime
    updated_by_run_id: str
    record_count: int
    latest_requested_at: datetime | None = None

File: frontend/src/types/ingestion.ts
export type IngestionRunAccepted = {
  runId: string;
  status: string;
};

export type IngestionRunStatus = {
  runId: string;
  status: string;
  resultType: string | null;
  startedAt: string;
  completedAt: string | null;
  cursorUsed: string | null;
  cursorAdvanced: boolean;
  candidateDatasetId: string | null;
  datasetVersionId: string | null;
  recordsReceived: number | null;
  failureReason: string | null;
};

export type CurrentDataset = {
  sourceName: string;
  datasetVersionId: string;
  updatedAt: string;
  updatedByRunId: string;
  recordCount: number;
  latestRequestedAt: string | null;
};
